### ISSUE TITLE
Not useable newspaper migration site

### ISSUE THREAD

**Issue by henning-gerhardt** [kitodo/kitodo-production#4384]
If you have a large database with a lot of newspaper processes (over 300,000) and corresponding batches (> 150) using the newspaper migration site is not possible as it takes hours or days  to display this page. While loading this page a lot of SELECT statements - which is ok - but even the same amount of UPDATE statements are started. It is for me unclear why UPDATE statements are used / fired on displaying this page.

Displaying the batches tab on the process page takes one or two minutes and is much faster then displaying the nearly same data on the newspaper migration site.

**Comment by matthias-ronge** [comment]
Can you see _what_ is updated? Process? Project? Batch?

**Comment by henning-gerhardt** [comment]
I can only confirm, that this long loading of this page (takes hours on our migration test system) appears only once and later access of this page is more or less fast - at least faster than on first access. But as migration of our data is not done yet I will struggle in this issue after every data migration :-(

I discovered that storing the changed value in the method `storeObject` of class [BaseDAO](https://github.com/kitodo/kitodo-production/blob/master/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/BaseDAO.java#L347-L360) take a lot of time: up to 15 seconds.

**Comment by matthias-ronge** [comment]
> if a non-null value is needed in `processBaseUri` and is set to the used process id, why did this is not done in database migration?

I assume it was because, when this was implemented, it was first a _blob_ field and these cannot be set by SQL language. So the getter obtains the value from the file management module when it is _null_, and saves it to the database.

But this is an idea, you should can set the field manually in the database, like:
```sql
UPDATE process SET processBaseUri = CONCAT(id, "/") WHERE processBaseUri IS NULL;
```

Another consideration is whether we should allow the file system check to be switched off via the `kitodo_config.properties`. Then all batches are displayed, including those that do not contain newspapers, but it would be much faster. Unfortunately, the route via the filemanagement module is very slow.

**Comment by henning-gerhardt** [comment]
Your suggestion is wrong or not correct as in database only a string representation of the process is stored without the ending slash! Even with the ending slash I don't understand the benefit of storing the process id with an ending slash. Accessing this field from the database has a higher cost then combining this two values in program code - only true if the process id is already available if not costs are similar.

Issue is not proving the existing of a file which is working really fast the remaining issue is to update the `processBaseUri` field which takes a long time on accessing this page.

**Comment by matthias-ronge** [comment]
> Your suggestion is wrong or not correct as in database only a string representation of the process is stored without the ending slash!

This is what I used to know, too, yes, but I checked my database and on my local system it is stored _with_ an ending slash. I was really surprised to see that this morning, because I also thought it would be stored without slash. The question is, why is there a difference? Are there are more methods that write this entry and generate different values? I have no idea where this comes from. If it is without slash, the database statement to set it is trivial.

> Even with the ending slash I don't understand the benefit of storing the process id with an ending slash.
I think this needs clarification. When I create a process in version 3, the field is set and it is _with_ slash.

> Issue is not proving the existing of a file which is working really fast
This is good to know, I wasn’t shure about it.
